pkg/ginkgo_helper/metrics.go

```diff
@@ -0,0 +1,113 @@
+package ginkgo_helper
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/format"
+	"github.com/pingcap/test-infra/sdk/resource"
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+func BuildPrometheusAPI(tc resource.TiDBCluster) (v1.API, error) {
+	url, err := tc.ServiceURL(resource.Prometheus)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := api.NewClient(api.Config{
+		Address: url.String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return v1.NewAPI(client), nil
+}
+
+type PromQLInstance struct {
+	description string
+	query       string
+	api         *v1.API
+}
+
+func PromQL(description string, query string, api *v1.API) PromQLInstance {
+	return PromQLInstance{
+		description: description,
+		query:       query,
+		api:         api,
+	}
+}
+
+type promQLEvaluatedMatcher struct {
```

Comment:
Can we seperate the gomega logic from promQL check logic?

Reply:
the main PromQL logic is only a small function, roughly 15 lines of code (`pkg/ginkgo_helper/metrics.go:74`). I think keep the function inplace could make it more compact and more readable.

Reply:
LGTM.